modules/@apostrophecms/admin-bar/index.js

```diff
@@ -14,6 +14,56 @@ module.exports = {
     pageTree: true
   },
   commands(self) {
+    const devicePreviewModeScreens = (
+      self.apos.asset.options.devicePreviewMode?.enable &&
+      self.apos.asset.options.devicePreviewMode?.screens
+    ) || {};
+    const devicePreviewModeCommands = {
+      [`${self.__meta.name}:toggle-device-preview-mode:exit`]: {
+        type: 'item',
+        label: {
+          key: 'apostrophe:commandMenuToggleDevicePreviewMode',
+          device: '$t(apostrophe:devicePreviewExit)'
+        },
+        action: {
+          type: 'command-menu-admin-bar-toggle-device-preview-mode',
+          payload: {
+            mode: null,
+            width: null,
+            height: null
+          }
+        },
+        shortcut: 'P,0'
+      }
+    };
+    let index = 1;
+    for (const [ name, screen ] of Object.entries(devicePreviewModeScreens)) {
+      // Up to 9 shortcuts available
+      if (index === 9) {
+        break;
+      }
+
+      devicePreviewModeCommands[`${self.__meta.name}:toggle-device-preview-mode:${name}`] = {
+        type: 'item',
+        label: {
+          key: 'apostrophe:commandMenuToggleDevicePreviewMode',
+          device: `$t(${screen.label})`
+        },
+        action: {
+          type: 'command-menu-admin-bar-toggle-device-preview-mode',
+          payload: {
+            mode: name,
+            label: `$t(${screen.label})`,
+            width: screen.width,
+            height: screen.height
+          }
+        },
+        shortcut: `P,${index}`
```

Comment:
```suggestion
        shortcut: `P${index}`
```

seems more UX-friendly to me

Reply:
will not work with current keyboard shortcut listener implementation

`P+0` is supposed to be `P` `and` `0`, but it comes with additional challenges (the order you're pressing the keys is important `P+0` and `0+P` should be different shortcuts)

`P,0` is translated to `P` `then` `0` (you still can hold `P` while pressing `0`), I think this is what you're looking for.